File: Old Python/IncreasingArray.py
import sys
def main():
    n = int(input())
    arr = list(map(int, input().split()))    
    diff = 0
    prev = arr[0]
    for i in range(1, len(arr)):
        cur = max(arr[i], prev)
        diff += max(0, cur - arr[i])
        prev = cur
    print(diff)
main()

File: IncreasingArray.cpp
#include <iostream>
typedef long long ll;
using namespace std;

int main() {
    int n;
    int prev;
    int cur;
    ll incs = 0;
    cin >> n;
    cin >> prev;
    for (int i = 0; i < n - 1; ++i) {
        cin >> cur;
        incs += max(0, prev - cur);
        prev = max(prev, cur);
    }
    cout << incs;
}
